pkg/domainservice/validateDependenciesDomainUseCase.go

```diff
@@ -0,0 +1,106 @@
+package domainservice
+
+import (
+	"errors"
+	"fmt"
+	"github.com/cloudogu/cesapp-lib/core"
+	"github.com/cloudogu/k8s-blueprint-operator/pkg/domain"
+	"github.com/cloudogu/k8s-blueprint-operator/pkg/util"
+)
+
+type ValidateDependenciesDomainUseCase struct {
+	remoteDoguRegistry RemoteDoguRegistry
+}
+
+func NewValidateDependenciesDomainUseCase(remoteDoguRegistry RemoteDoguRegistry) *ValidateDependenciesDomainUseCase {
+	return &ValidateDependenciesDomainUseCase{
+		remoteDoguRegistry,
+	}
+}
+
+// ValidateDependenciesForAllDogus checks if for all dogus in the blueprint there are also all dependencies in the blueprint.
+// The dependencies are validated against dogu specifications in a remote dogu registry.
+// This functions returns no error if everything is ok or
+// a domain.InvalidBlueprintError if there are dependencies missing or
+// an InternalError if there is any other error, e.g. with the connection to the remote dogu registry
+func (useCase *ValidateDependenciesDomainUseCase) ValidateDependenciesForAllDogus(effectiveBlueprint domain.EffectiveBlueprint) error {
+	wantedDogus := effectiveBlueprint.GetWantedDogus()
+	dogusToLoad := util.Map(wantedDogus, func(dogu domain.Dogu) DoguToLoad {
+		return DoguToLoad{
+			QualifiedDoguName: dogu.GetQualifiedName(),
+			Version:           dogu.Version.Raw,
+		}
+	})
+	doguSpecsOfWantedDogus, err := useCase.remoteDoguRegistry.GetDogus(dogusToLoad)
+	if err != nil {
+		var notFoundError *NotFoundError
+		if errors.As(err, &notFoundError) {
+			return &domain.InvalidBlueprintError{WrappedError: err, Message: "remote dogu registry has no dogu specification for at least one wanted dogu"}
+		} else { //should be InternalError
+			return &InternalError{WrappedError: err, Message: "cannot load dogu specifications from remote registry for dogu dependency validation"}
+		}
+	}
+
+	for _, wantedDogu := range wantedDogus {
+		dependencyDoguSpec := doguSpecsOfWantedDogus[wantedDogu.GetQualifiedName()]
+		err = useCase.checkDoguDependencies(wantedDogus, doguSpecsOfWantedDogus, dependencyDoguSpec.Dependencies)
+		if err != nil {
+			err = fmt.Errorf("dependencies for dogu '%s' are not satisfied in blueprint: %w", wantedDogu.Name, err)
+		}
+	}
+
+	return err
```

Comment:
This only returns the last error. Is this intended? Shouldn't these errors be collected and joined?

Reply:
This is a serious bug. I committed a fix and a test for this case. Thank you very much!